miprometheus/problems/question_context_to_class/babiqa_dataset_single_question.py

```diff
@@ -0,0 +1,514 @@
+
+            # go through all the stories
+            for answer in story:
+                a = answer.lower()
+                if a not in answ_to_ix:
+                    ix = len(answ_to_ix)
+                    answ_to_ix[a] = ix
+                    itos_d.append(a)
+                    # print(a,ix)
+
+            # go through all the answers
+            for answer in answers:
+                a = answer.lower()
+                if a not in answ_to_ix:
+                    ix = len(answ_to_ix)
+                    answ_to_ix[a] = ix
+                    itos_d.append(a)
+                    # print(a, ix)
+
+        """ build embeddings from the chosen database / Example: glove.6B.100d """
+
+        self.language.build_pretrained_vocab(text, vectors=self.embedding_type, tokenize=self.tokenize)
+
+        # return the corresponding dictionnaries
+        ret = (answ_to_ix)
+        return ret, itos_d
+
+
+
+    def download_from_url(self, url, path):
+
+        """Download file, with logic (from tensor2tensor) for Google Drive"""
+
+        #get url and write file to path
+        if 'drive.google.com' not in url:
+            r = requests.get(url, headers={'User-Agent': 'Mozilla/5.0'})
+            with open(path, "wb") as file:
+                file.write(r.content)
+            return
+        print('downloading from Google Drive; may take a few minutes')
+        confirm_token = None
+        session = requests.Session()
+        response = session.get(url, stream=True)
+        for k, v in response.cookies.items():
+            if k.startswith("download_warning"):
+                confirm_token = v
+
+        if confirm_token:
+            url = url + "&confirm=" + confirm_token
+            response = session.get(url, stream=True)
+
+        #open file + write chunks
+        chunk_size = 16 * 1024
+        with open(path, "wb") as f:
+            for chunk in response.iter_content(chunk_size):
+                if chunk:
+                    f.write(chunk)
+
+    def load_data(self, path=None, root='.data', tasks=[1], tenK=False, add_punctuation=True, data_type='train',
+                  outmod=''):
+
+        """loads all asked for tasks into a single file (combining multiple files) and then parses the combined file"""
+
+        if tenK:
+            self.dirname = os.path.join('tasks_1-20_v1-2', 'en-valid-10k')
+        else:
+            self.dirname = os.path.join('tasks_1-20_v1-2', 'en-valid')
+
+        if path is None:
+            path = self.download(root)
+
+        file_data = os.path.join(path, 'collected_' + data_type + outmod + '.txt')
+        with open(file_data, 'w') as tf:
+            for task in tasks:
+                with open(
+                        os.path.join(path,
+                                     'qa' + str(task) + '_' + data_type + '.txt')) as f:
+                    tf.write(f.read())
+        return self.parse(file_data, add_punctuation)
+
+    def download(self, root, check=None):
```

Comment:
Ok, so that whole part should at one point be integrated with Emre's ProblemInitializer

Reply:
Why 2 `download` methods?

Reply:
One downloads single file from a given url, the second one downloads (using the first one) file(s?), unpacks it and returns a path to it. IMO this is really 100% ProblemInitializer functionality

Reply:
Ok I will to know more about that. Will check with Emre